src/www/app/outline_server_repository/index.ts

```diff
@@ -17,26 +17,41 @@ import uuidv4 from 'uuidv4';
 
 import * as errors from '../../model/errors';
 import * as events from '../../model/events';
-import {ServerRepository} from '../../model/server';
+import {ServerRepository, ServerType} from '../../model/server';
 
 import {NativeNetworking} from '../net';
 import {TunnelFactory} from '../tunnel';
 
 import {OutlineServer} from './server';
-import {accessKeyToShadowsocksSessionConfig} from './access_key_serialization';
+import {staticKeyToShadowsocksSessionConfig} from './access_key_serialization';
 
 // Compares access keys proxying parameters.
-function accessKeysMatch(a: string, b: string): boolean {
+function staticKeysMatch(a: string, b: string): boolean {
   try {
-    const l = accessKeyToShadowsocksSessionConfig(a);
-    const r = accessKeyToShadowsocksSessionConfig(b);
+    const l = staticKeyToShadowsocksSessionConfig(a);
+    const r = staticKeyToShadowsocksSessionConfig(b);
     return l.host === r.host && l.port === r.port && l.password === r.password && l.method === r.method;
   } catch (e) {
     console.debug(`failed to parse access key for comparison`);
   }
   return false;
 }
 
+// Determines if the key is expected to be a url pointing to an ephemeral session config.
+function isDynamicAccessKey(accessKey: string): boolean {
+  return accessKey.startsWith('ssconf://') || accessKey.startsWith('https://');
+}
+
+// NOTE: For extracting a name that the user has explicitly set, only.
+// (Currenly done by setting the hash on the URI)
+function serverNameFromAccessKey(accessKey: string): string {
```

Comment:
It would be great to have unit tests for these functions.

Reply:
also added a TODO for this